api/proto/auth/config.go

```diff
@@ -0,0 +1,97 @@
+package auth
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"time"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/trace"
+)
+
+// Config contains configuration of the client
+type Config struct {
+	// Addrs is a list of addresses to dial
+	Addrs []string
+	// Dialer is a custom dialer, if provided
+	// is used instead of the list of addresses
+	Dialer ContextDialer
+	// KeepAlivePeriod defines period between keep alives
+	KeepAlivePeriod time.Duration
+	// KeepAliveCount specifies amount of missed keep alives
+	// to wait for until declaring connection as broken
+	KeepAliveCount int
+	// TLS is a TLS config
+	TLS *tls.Config
+}
+
+// CheckAndSetDefaults checks and sets default config values
+func (c *Config) CheckAndSetDefaults() error {
+	if len(c.Addrs) == 0 && c.Dialer == nil {
+		return trace.BadParameter("set parameter Addrs or DialContext")
+	}
+	if c.TLS == nil {
+		return trace.BadParameter("missing parameter TLS")
+	}
+	if c.KeepAlivePeriod == 0 {
+		c.KeepAlivePeriod = ServerKeepAliveTTL
+	}
+	if c.KeepAliveCount == 0 {
+		c.KeepAliveCount = KeepAliveCountMax
+	}
+	if c.Dialer == nil {
+		c.Dialer = NewAddrDialer(c.Addrs, c.KeepAlivePeriod)
+	}
+	if c.TLS.ServerName == "" {
+		c.TLS.ServerName = teleport.APIDomain
+	}
+	// this logic is necessary to force client to always send certificate
```

Comment:
IIRC we've established that this clutch is no longer needed, but I'm not 100% sure, @awly or @russjones  may remember better.